### PR TITLE
Using 'managed'/'unmanaged' everywhere

### PR DESCRIPTION
**Pull request by austinzheng**
This should be the last copyediting/terminology PR. Thanks for your time.

@bdash @tgoyne @jpsim

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -629,7 +629,7 @@ Prebuilt frameworks are now built with Xcode 7.1.
 
 ### Bugfixes
 
-* Setting the primary key property on persisted `RLMObject`s / `Object`s
+* Setting the primary key property on managed `RLMObject`s / `Object`s
```
Comment (austinzheng) [inline CHANGELOG.md]: Is this also true for computed properties?

File: CHANGELOG.md [hunk 2]
```diff
@@ -689,7 +689,7 @@ Prebuilt frameworks are now built with Xcode 7.1.
 
 ### Enhancements
 
-* Persisted properties of `RLMObject`/`Object` subclasses are now Key-Value
+* Managed properties of `RLMObject`/`Object` subclasses are now Key-Value
```
Comment (austinzheng) [inline CHANGELOG.md]: Is this also true for computed properties?

File: CHANGELOG.md (hunk 1 above)
Comment (bdash) [inline CHANGELOG.md]: Computed properties cannot be set and also cannot be used as the primary key.

**Comment by jpsim** [comment]
I'd rather not modify older parts of the changelog @austinzheng. Otherwise these changes look fine to me. Don't forget to update the cocoa docs in the website repo too!

**Comment by tgoyne** [comment]
I agree on not changing the changelog.

The new wording is incorrect for `createInRealm:withValue:` &c. RLMLinkingObjects/LinkingObjects properties are managed properties, but won't be used from input dictionaries/other KVC objects and cannot appear in input arrays. I don't have any great ideas for how to update the wording to fix this.

**Comment by austinzheng** [comment]
What about this for a docstring?

```
/**
 Creates an instance of a Realm object with a given value, and adds it to the specified Realm.
 
 If nested objects are included in the argument, `createInRealm:withValue:` will be recursively called
 on them.
 
 The `value` argument can be any key-value coding compliant object, an array or dictionary returned from the methods in
 `NSJSONSerialization`, or an `NSArray` containing one element for each managed property. An exception will be thrown if
 any required properties are not present and those properties were not defined with default values. Do not pass in a
 `RLMLinkingObjects` instance, either by itself or as a member of a collection.
 
 When passing in an `NSArray` as the `value` argument, all properties must be present, valid and in the same order as
 the properties defined in the model.

 @param realm    The Realm which should manage the newly-created object.
 @param value    The value used to populate the object.

 @see   `defaultPropertyValues`
 */
```

As far as I can tell it's impossible to give a `param` a multi-paragraph description.

**Comment by jpsim** [comment]
> As far as I can tell it's impossible to give a `param` a multi-paragraph description.

Correct, in both Swift and ObjC, since Jazzy renders doc comments as Markdown, converting `@param` to unordered list items, and AFAICT in Markdown you can't have multiple paragraphs within a single bullet list item.

**Comment by jpsim** [comment]
There are trailing spaces in the Swift files FYI, causing SwiftLint violations.

**Comment by austinzheng** [comment]
Are you okay with the rewritten description of `value`? I want to make sure it's not misleading.

**Comment by jpsim** [comment]
Could you please fix the merge issues here, I'd be happy to take another look afterwards.

**Comment by austinzheng** [comment]
I've rebased onto master. Would appreciate a review whenever you have time to spare. @bdash @tgoyne @jpsim

**Comment by mrackwitz** [comment]
This makes still retrospective modifications to the changelog. Shouldn't those be reverted?

File: Realm/RLMMigration.h [hunk 3]
```diff
@@ -75,15 +75,17 @@ typedef void (^RLMObjectMigrationBlock)(RLMObject * __nullable oldObject, RLMObj
 
 /**
  Creates and returns an `RLMObject` instance of type `className` in the Realm being migrated.
+ 
+ The `value` argument is used to populate the object. It can be a key-value coding compliant object, an array or 
+ dictionary returned from the methods in `NSJSONSerialization`, or an `NSArray` containing one element for each managed
+ property. An exception will be thrown if any required properties are not present and those properties were not defined
+ with default values. Do not pass in a `RLMLinkingObjects` instance, either by itself or as a member of a collection.
```
Comment (jpsim) [inline Realm/RLMMigration.h]: I don't understand how this last sentence is useful.

File: Realm/RLMMigration.h [hunk 4]
```diff
@@ -75,15 +75,17 @@ typedef void (^RLMObjectMigrationBlock)(RLMObject * __nullable oldObject, RLMObj
...
+
+ When passing in an `NSArray` as the `value` argument, all properties must be present, valid and in the same order as
```
Comment (jpsim) [inline Realm/RLMMigration.h]: we interchangeably use "`NSArray`" or "array" in this comment. Let's stick with "array".

**Comment by jpsim** [comment]
Just need to revert changes to changelog and address my two small comments, then 👍 from me.

**Comment by austinzheng** [comment]
I'm going to squash, because not doing so makes reverting stuff too difficult.

**Comment by austinzheng** [comment]
Does anyone have any objection to this API documentation for the following method (and others like it)?

```objective-c
 Creates an `RLMObject` instance of type `className` in the Realm, and populates it using a given object.
 
 The `value` argument is used to populate the object. It can be a key-value coding compliant object, an array or
 dictionary returned from the methods in `NSJSONSerialization`, or an array containing one element for each managed
 property. An exception will be thrown if any required properties are not present and those properties were not defined
 with default values.

 When passing in an array as the `value` argument, all properties must be present, valid and in the same order as the
 properties defined in the model.

 @warning This method is useful only in specialized circumstances, for example, when building components
          that integrate with Realm. If you are simply building an app on Realm, it is recommended to
          use `[RLMObject createInDefaultRealmWithValue:]`.

 @param value    The value used to populate the object.

 @return    An `RLMObject` instance of type `className`.
 */
-(RLMObject *)createObject:(NSString *)className withValue:(id)value;
```

**Comment by jpsim** [comment]
Your last comment seems fine @austinzheng

**Comment by austinzheng** [comment]
Please file issues against me if you want any more documentation changes.

**Pull request merged**